controllers/appmesh/cloudmap_controller.go

```diff
@@ -83,7 +89,11 @@ func (r *cloudMapReconciler) reconcile(ctx context.Context, req ctrl.Request) er
 	if !vNode.DeletionTimestamp.IsZero() {
 		return r.cleanupCloudMapResources(ctx, vNode)
 	}
-	return r.reconcileVirtualNodeWithCloudMap(ctx, vNode)
+	if err := r.reconcileVirtualNodeWithCloudMap(ctx, vNode); err != nil {
+		r.recorder.Event(vNode, corev1.EventTypeWarning, "ReconcileError", err.Error())
```

Comment:
This one is pretty important. We have a bug related to this. Do you know if the inner function (reconcileVirtualNodeWithCloudmap) or the caller log the details of the failure ?

Reply:
@cgchinmay If the reconcile call fails, where does the retry happen ? I ask because do we want to guard against an Event storm if retries are done at a higher level.

Reply:
From what I know, the controller-runtime invokes the Reconcile loop every time we return an error or for any of the create, update, delete events based on how we have configured the watcher in SetupWithManager. It does have a built-in back off logic. So if we want to stop the reconcile loop then I guess we need to return nil. We can decide this based on the error type whether it's intermittent or not, since doing it based on number of retries could be hard. Will have to test this logic.